exercises/diffie-hellman/src/test/java/DiffieHellmanTest.java

```diff
@@ -0,0 +1,95 @@
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+import java.math.BigInteger;
+import java.util.*;
+
+public class DiffieHellmanTest {
+
+    private DiffieHellman diffieHellman;
+
+    @Before
+    public void setUp() {
+        diffieHellman = new DiffieHellman();
+    }
+
+    @Test
+    public void testPrivateKeyInRange() {
+        BigInteger primeP = BigInteger.valueOf(23);
+        List<BigInteger> privateKeys = new ArrayList<>();
+        for (int i = 0; i < 10; i++) privateKeys.add(diffieHellman.privateKey(primeP));
+
+        for (BigInteger privateKey : privateKeys) {
+            assertTrue((privateKey.compareTo(BigInteger.ONE) >= 0) && (privateKey.compareTo(primeP) < 0));
+        }
+    }
+
+    // Note: due to the nature of randomness, there is always a chance that this test fails
```

Comment:
Not sure it would be ideal to have a test that could fail even if the implementation is correct.

This could cause issues with builds, and give us some grief. Could you explain to me why this test intermittently fails?

Reply:
5 different numbers are being generated (in a range of 0 to 7919) and we are checking that they are different to ensure that ```privateKey``` function generates random numbers. There is a very small chance of getting some number repeated and fail tests.

Reply:
There's a similar problem in the [robot-name tests](https://github.com/exercism/java/blob/master/exercises/robot-name/src/test/java/RobotTest.java). In that case the problem description says each robot should have a different name but most solutions will probably use randomness in which case there's a chance the names will be the same.

I think it's fine to keep this test, but I would only generate two keys and assert that they're not the same. That makes it a bit more obvious what you're testing 🙂

Reply:
OK, I also think two keys will be enough 🙂